src/libsystemd/sd-event/test-event.c

```diff
@@ -590,6 +591,57 @@ static void test_pidfd(void) {
         sd_event_unref(e);
 }
 
+static int ratelimit_handler(sd_event_source *s, int fd, uint32_t revents, void *userdata) {
+        unsigned *c = (unsigned*) userdata;
+        *c += 1;
+        return 0;
+}
+
+static void test_ratelimit(void) {
+        sd_event *e = NULL;
+        sd_event_source *t = NULL;
+        int p[2] = {-1, -1};
+        uint64_t burst, interval;
+        unsigned i, count;
+
+        assert_se(sd_event_default(&e) >= 0);
+        assert_se(pipe(p) == 0);
```

Comment:
hmm, please use pipe2() with O_CLOEXEC.

It doesn't really matte rin this specific case, but our coding style says we should never create fds with O_CLOEXEC not set unless there's a really good reason for it. sooner or later some static checker/coccinelle would notice